docs/config_search.md

```diff
@@ -276,6 +280,106 @@ profile_models:
 
 ---
 
+## Optuna Search Mode
+
+**-ALPHA RELEASE-**
+
+_This mode has the following limitations:_
+
+- **Ensemble, BLS or concurrent multi-model profiling is not supported**
+- **Profiling with request rate is not supported**
+
+This mode uses a hyperparameter optimization framework to search the configuration
+space, looking for the maximal objective value within the specified constraints.
+Please see the [Optuna](https://optuna.org/) website if you are interested in specific details on how the algorithm functions.
+
+Optuna allows you to search for every parameter that can be specified in the model configuration. Parameters can be specified
+with a min/max range (using the run-config-search options) or a list of parameters to test against can be set in the
+parameters/model_config_parameters field.
+
+After optuna search has found the best config(s), it will then sweep the top-N configurations found (specified by `--num-configs-per-model`) over the default concurrency range before generation of the summary reports.
+
+---
+
+_An example model analyzer YAML config that performs an Optuna Search:_
+
+```yaml
+model_repository: /path/to/model/repository/
+
+run_config_search_mode: optuna
+profile_models:
+  - model_A
+```
+
+A number of new configuration options were added to support tailoring the Optuna search to your needs:
+
+- `--min/max_percentage_of_search_space`: sets the percentage of the space you want Optuna to search
+- `--optuna-min/max-trials`: sets the number of trials Optuna will attempt
+- `--optuna-early-exit-threshold`: sets the number of trials without improvement before triggering early exit
+- `--use-concurrency-formula`: uses a formula (2 \* batch size \* instance group count), rather than sweeping concurrency
+
+---
+
+_An example that performs an Optuna Search using these new configuration options:_
+
+```yaml
+model_repository: /path/to/model/repository/
+
+run_config_search_mode: optuna
+run_config_search_max_instance_count: 8
+run_config_search_min_concurrency: 32
+run_config_search_max_concurrency: 256
+
+use_concurrency_formula: True
+min_percentage_of_search_space: 10
+optuna_max_trials: 200
+optuna_early_exit_threshold: 15
+
+profile_models:
+  model_A:
+    model_config_parameters:
+      max_batch_size: [1, 4, 8, 32, 64, 128]
+      dynamic_batching:
+        max_queue_delay_microseconds: [100, 200, 300]
+    parameters:
+      batch_sizes: 1, 2, 4, 8, 16
+```
+
+_The debug output showing how the space will be searched:_
+
+```yaml
+Number of configs in search space: 720
+   batch_sizes: [1, 2, 4, 8, 16] (5)
```

Comment:
[tgerdesnv](https://github.com/tgerdesnv) [26 minutes ago](https://github.com/triton-inference-server/model_analyzer/pull/894#discussion_r1629507062)
Is there an example that isn't all the options we already support?

Member
Author
@[nv-braf](https://github.com/nv-braf) nv-braf [16 minutes ago](https://github.com/triton-inference-server/model_analyzer/pull/894#discussion_r1629522786)
This example uses options that quick search doesn't support (and has enumerated lists).